src/lib/net/inaddr.c

```diff
@@ -39,8 +39,31 @@ tor_inet_aton(const char *str, struct in_addr *addr)
 {
   unsigned a,b,c,d;
   char more;
+  char octet[4];
+  size_t octet_id = 0;
   if (tor_sscanf(str, "%3u.%3u.%3u.%3u%c", &a, &b, &c, &d, &more) != 4)
     return 0;
+
+  /* Parse the octets and check them for leading zeros. This avoids using the
+   * heap.
+   *
+   * Here, we run until idx <= strlen(str) so we don't skip the last character
+   * and therefore can process it carefully. However, we cannot do a buffer
+   * overread because idx == strlen(str) does not read further.
+   *
+   * Also, the tor_sscanf() call above prevents an overflow from occuring. */
+  for (size_t idx = 0; idx <= strlen(str); idx++) {
+    if (str[idx] == '.' || idx == strlen(str)) {
+      if (strlen(octet) > 1 && octet[0] == '0')
```

Comment:
`octet` contains junk at the start of the loop so what happens if we end up here at the first iteration where `octet` was never set?

Reply:
The tor_sscanf() check should prevent this.

Reply:
Ok I think I know what you mean as in we can only get in that `if()` with a malformed address and the sscanf doesn't allow that.

I would really love to still do have `char octet[4] = {0};` just for safety and future proofing this. As in setting it to zeroes at first.